src/aws/step-function/generateStateMachineParams.js

```diff
@@ -1,29 +1,25 @@
-const { iam } = require("../services");
+const aslValidator = require("asl-validator");
 const fs = require("fs");
+const { iam } = require("../services");
 const workflowName = require("../../util/workflowName");
-const { accountNumber, region } = require("../../util/awsAccountInfo");
-
-const replacePlaceholders = (definitionTemplate) => {
-  let definition = definitionTemplate;
-
-  definition = definition.replace(/REGION/g, region);
-  definition = definition.replace(/ACCOUNT_ID/g, accountNumber);
-  definition = definition.replace(/WORKFLOW_NAME/g, workflowName);
-
-  return definition;
-};
+const replacePlaceholders = require("../../util/replacePlaceholders");
 
 const generateStateMachineParams = async (roleName) => {
   const role = await iam.getRole({ RoleName: roleName }).promise();
-  const definition = replacePlaceholders(
-    fs.readFileSync("definition.asl.json").toString()
-  );
+  const aslTemplate = fs.readFileSync("definition.asl.json").toString();
+  const definition = replacePlaceholders(aslTemplate);
+  const { isValid, errorsText } = aslValidator(JSON.parse(definition));
 
-  return {
-    definition,
-    name: workflowName,
-    roleArn: role.Role.Arn,
-  };
+  if (isValid) {
+    return {
+      definition,
+      name: workflowName,
+      roleArn: role.Role.Arn,
+    };
+  } else {
+    console.error("✕ State machine definition is invalid:", errorsText("\n"));
+    return;
```

Comment:
This is good, but the SDK still tries to create the state machine, just with an empty definition instead.

Reply:
There should probably be some validation in the `src/aws/step-function/createStepFunction.js` file. Maybe this logic should be moved there?

Reply:
Good point. I've moved the logic to `createStepFunctions.js` instead.